### ISSUE TITLE
Change the way stacks of items are displayed

### ISSUE THREAD

**Issue by LeTacoBot** [CleverRaven/Cataclysm-DDA#18245]
Similar items stack really nicely in the inventory:
![arbitrary_alt_text](http://i.imgur.com/yqmgHTS.png)

But not so cleanly when they're in a pile on the ground:
![more_arbitrary_alt_text](http://i.imgur.com/ExYWtPA.png)

I may be missing some reason why this should stay, but it'd be nice if it were easier to leaf through huge stockpiles without getting into the " `w " range.

**Comment by BevapDin** [comment]
Looks like a duplicate of #2531.

**Comment by sethsimon** [comment]
I hope to look into this once #18195 is finished. It would make sense to move the advanced inventory's hand-rolled `item_stack` into the `item_stack` class somehow.

**Comment by nornagon** [comment]
I've been poking around in the pickup code to get my hands dirty and try to fix this. One thing I came across is an issue when trying to pick up e.g. 200 rags, when that's more than fits in your inventory. When you select that stack, should it just show volume as red and prevent the pickup? Should it prompt for how many you want to pick up (a la AIM)? Should it just select the largest number (smaller than "all of them") that would fit in your inventory?

**Comment by LeTacoBot** [comment]
I was thinking exactly like the inventory, where it's "all" by default, and you can type in a number to pick up a certain amount. However, that might be a little complicated.

**Comment by nornagon** [comment]
The pickup UI already handles typing a number to select less than all of the items, but currently it only applies to items with charges (e.g. batteries). The way it currently works is that if you've selected any items that would put your inventory over the volume limit, it asks if you want to wield them instead. If you choose more than one such item, it'll ask you to wield the first one, then ask you to drop it and wield the second one instead, and so forth.

That seems like a reasonable way to handle the same edge case for non-ammo items, too, so I'd say it should stay that way :)

**Comment by nornagon** [comment]
I have a branch that fixes this: https://github.com/nornagon/Cataclysm-DDA/tree/pickup-stacks

But it depends on #18280 and I don't know how to do "dependent" pull requests on GitHub.

**Comment by Zireael07** [comment]
> I don't know how to do "dependent" pull requests on GitHub.

I think making a branch from the other branch would do it.

**Comment by mugling** [comment]
> I don't know how to do "dependent" pull requests on GitHub.
> 
> I think making a branch from the other branch would do it.

Let me take a look at your first PR as this unblocks your progress
